src/sanctum/src/Http/Middleware/EnsureFrontendRequestsAreStateful.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Hypervel\Sanctum\Http\Middleware;
+
+use Hyperf\Collection\Collection;
+use Hyperf\HttpServer\Contract\RequestInterface;
+use Hyperf\HttpServer\Contract\ResponseInterface as HttpResponse;
+use Hypervel\Support\Str;
+use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class EnsureFrontendRequestsAreStateful implements MiddlewareInterface
+{
+    /**
+     * The middleware to apply to stateful requests.
+     *
+     * @var array<class-string>
+     */
+    protected array $frontendMiddleware = [];
+
+    public function __construct(
+        protected ContainerInterface $container,
+        protected RequestInterface $request,
+        protected HttpResponse $response
+    ) {
+        $this->configureFrontendMiddleware();
+    }
+
+    /**
+     * Process the incoming request.
+     */
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $this->configureSecureCookieSessions();
+
+        if (static::fromFrontend($this->request)) {
+            // Mark request as stateful
+            $request = $request->withAttribute('sanctum', true);
+
+            // Apply frontend middleware chain
+            return $this->applyFrontendMiddleware($request, $handler);
+        }
+
+        return $handler->handle($request);
+    }
+
+    /**
+     * Configure secure cookie sessions.
+     */
+    protected function configureSecureCookieSessions(): void
+    {
+        config([
+            'session.http_only' => true,
+            'session.same_site' => 'lax',
+        ]);
+    }
+
+    /**
+     * Configure the middleware that should be applied to requests from the "frontend".
+     */
+    protected function configureFrontendMiddleware(): void
+    {
+        $this->frontendMiddleware = array_values(array_filter([
+            config('sanctum.middleware.add_queued_cookies'),
+            config('sanctum.middleware.start_session'),
+            config('sanctum.middleware.verify_csrf_token'),
+            config('sanctum.middleware.authenticate_session'),
+        ]));
+    }
+
+    /**
+     * Apply the frontend middleware to the request.
+     */
+    protected function applyFrontendMiddleware(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        // Build middleware chain in reverse order
+        $next = $handler;
+
+        foreach (array_reverse($this->frontendMiddleware) as $middleware) {
+            if (! $middleware) {
+                continue;
+            }
+
+            $next = new class($this->container, $middleware, $next) implements RequestHandlerInterface {
```

Comment:
1. we don't need to new middleware in each request?
2. we need to support PSR middleware and Laravel-style middleware at the same time.

We can use HTTP dispatcher's pipeline here, you can refer to:  https://github.com/hypervel/components/blob/main/src/dispatcher/class_map/HttpRequestHandler.php#L27-L31